test/remote-test.js

```diff
@@ -1873,3 +1960,499 @@ describe('Remote', function() {
     });
   });
 });
+
+describe.skip('Request API consistency tests', function() {
+  // XXX convert all this to use fixtures
+
+  // Method parameters may be translated into multiple request parameters;
+  // 'ledger' may become the request param 'ledger_index' or 'ledger_hash'
+  // depending on the argument type.
+  //
+  // This function maps an args array to a new array that only includes one
+  // form of request parameter: the first specified
+  //
+  // ['ledger', [['ledger_index', LEDGER_INDEX], ['ledger_hash', LEDGER_HASH]]]
+  //
+  // should become
+  //
+  // ['ledger_index', LEDGER_INDEX]
+  function firstReqParam(arg) {
+    if (lodash.isEmpty(arg)) {
+      return undefined;
+    }
+    return Array.isArray(arg[1])
+    ? arg[1][0]
+    : arg;
+  }
+  assert.deepEqual(
+    firstReqParam(['ledger', [['ledger_index', LEDGER_INDEX], ['ledger_hash', LEDGER_HASH]]]),
+    ['ledger_index', LEDGER_INDEX]
+  );
+  assert.deepEqual(
+    firstReqParam(['account', ADDRESS]),
+    ['account', ADDRESS]
+  );
+
+  // ['ledger', [['ledger_index', LEDGER_INDEX], ['ledger_hash', LEDGER_HASH]]]
+  //
+  // should become
+  //
+  // ['ledger', LEDGER_INDEX]
+  function firstMethodParam(arg) {
+    if (lodash.isEmpty(arg)) {
+      return undefined;
+    }
+    return Array.isArray(arg[1])
+    ? [arg[0], arg[1][0][1]]
+    : arg;
+  }
+  assert.deepEqual(
+    firstMethodParam(['ledger', [['ledger_index', LEDGER_INDEX], ['ledger_hash', LEDGER_HASH]]]),
+    ['ledger', LEDGER_INDEX]
+  );
+  assert.deepEqual(
+    firstMethodParam(['account', ADDRESS]),
+    ['account', ADDRESS]
+  );
+
+  // Filters default params against optional params
+  function filterDefault(optionalArgs, defaultArg) {
+    return lodash.isEmpty(defaultArg)
+    || lodash.isEmpty(optionalArgs)
```

Comment:
I find it easier to read multi-line expression when lines after the first are indented.